packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+   */
+  maxSize?: number;
+  /**
+   * Additional class names to apply to the file input
+   */
+  className?: string;
+  /**
+   * If `true`, the file input will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * The success text content that will be shown below the input
+   */
+  successText?: string;
+  /**
+   * The error text content that will be shown below the input
+   */
+  errorText?: string;
+  /**
+   * The helper text content that will be shown below the input
+   */
+  helperText?: string;
+  /**
+   * The info text content that will be shown below the input
+   */
+  infoText?: string;
+  /**
+   * If `true`, the label is displayed as required and the `input` element will be required
+   */
+  required?: boolean;
+  /**
+   * Override or extend the styles applied to the component
+   */
+  style?: React.CSSProperties;
+};
+
+const formatBytes = (bytes: number): string => {
+  if (bytes === 0) {
+    return '0 bytes';
+  }
+
+  const sizeUnits: string[] = ['B', 'kB', 'MB', 'GB', 'TB'];
+  const sizeUnitIndex = Math.floor(Math.log(bytes) / Math.log(1024));
+  const sizeInUnit = bytes / 1024 ** sizeUnitIndex;
+  return `${sizeUnitIndex < 2 || sizeInUnit % 1 === 0 ? Math.round(sizeInUnit) : sizeInUnit.toFixed(1)} ${
+    sizeUnits[sizeUnitIndex]
+  }`;
+};
+
+const getNoFilesAddedMessage = (language: Language): string => {
+  return {
+    en: 'No file(s) selected.',
+    fi: 'Yhtään tiedostoa ei ole valittu.',
+    sv: 'No file(s) selected.',
+  }[language];
+};
+
+const getRemoveButtonLabel = (language: Language): string => {
+  return {
+    en: 'Remove',
+    fi: 'Poista',
+    sv: 'Ta bort',
+  }[language];
+};
+
+const getRemoveButtonAriaLabel = (language: Language, fileName: string): string => {
+  return {
+    en: `Remove ${fileName} from the added files list.`,
+    fi: `Poista tiedosto ${fileName} lisättyjen tiedostojen listasta.`,
+    sv: `Ta bort ${fileName} tillagda filen från fillistan.`,
+  }[language];
+};
+
+const getFileListAriaLabel = (language: Language, totalAddedFiles: number): string => {
+  return {
+    en: `${totalAddedFiles === 0 ? '1 file' : `${totalAddedFiles} files`} added.`,
+    fi: `${totalAddedFiles === 0 ? '1 tiedosto' : `${totalAddedFiles} tiedostoa`} added.`,
+    sv: `${totalAddedFiles === 0 ? '1 file' : `${totalAddedFiles} files`} added.`,
+  }[language];
+};
+
+const getRemoveSuccessMessage = (language: Language): string => {
+  return {
+    en: 'File removed.',
+    fi: 'Tiedosto poistettu.',
+    sv: 'Filen borttagen.',
+  }[language];
+};
+
+const getAddSuccessMessage = (language: Language, numberOfAdded: number, numberOfTotal: number): string => {
+  const partOfTotalStr = numberOfAdded === numberOfTotal ? numberOfAdded : `${numberOfAdded}/${numberOfTotal}`;
+
+  return {
+    en: `${partOfTotalStr} file(s) added.`,
+    fi: `${partOfTotalStr} tiedosto(a) lisätty.`,
+    sv: `${partOfTotalStr} file(s) added.`,
```

Comment:
Seems to be missing the swedish translation

Reply:
Yes, true. We have to remember to fix these 👍

Reply:
- [ ] fix translations